multicluster/controllers/multicluster/member_clusterset_controller.go

```diff
@@ -92,6 +92,9 @@ func (r *MemberClusterSetReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		}
 		klog.InfoS("Received ClusterSet delete", "clusterset", req.NamespacedName)
 		if r.remoteCommonArea != nil {
+			if err := r.deleteMemberClusterAnnounce(ctx); err != nil {
+				return ctrl.Result{}, err
```

Comment:
I think you need to prepend context to the returned err (or log the failure with context here), otherwise the log by the caller wont indicate what fails.

e.g. fmt.Errorf("failed to delete MemberClusterAnnounce in the leader cluster: %v", err)

Reply:
Fixed